### PR TITLE
DDF-4035: Add Policies and Enable Sec Manager for Standalone Solr

### PR DESCRIPTION
**Pull request by garrettfreibott**
2.13.x port: https://github.com/codice/ddf/pull/3626

#### What does this PR do?
Enables the Pro-Grade security manager on the JVM running the standalone Solr instance.

#### Who is reviewing it? 
@mcalcote 
@ahoffer 

#### Select relevant component teams: 
@codice/security 

#### Ask 2 committers to review/merge the PR and tag them here.
@coyotesqrl 
@stustison

#### How should this be tested?
Run itests and verify that all Solr functionality such and ingest, query, and retrieval still work as expected.

#### What are the relevant tickets?
[DDF-4053](https://codice.atlassian.net/browse/DDF-4035)

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: distribution/solr-distro/src/main/resources/bin/solr [hunk 1]
```diff
@@ -0,0 +1,2170 @@
...
+    esac
+  done
+fi
+
+if [[ $SOLR_LOG_LEVEL ]] ; then
+  SOLR_LOG_LEVEL_OPT="-Dsolr.log.level=$SOLR_LOG_LEVEL"
+fi
+
+if [ -z "$SOLR_SERVER_DIR" ]; then
+  SOLR_SERVER_DIR
```
Comment (garrettfreibott) [inline distribution/solr-distro/src/main/resources/bin/solr]: the `-D*` flags were added to the foreground and background `exec` commands. The `-jar start.jar` was also replaced with `-classpath ../security/pro-grade-1.1.3.jar:start.jar org.eclipse.jetty.start.Main` in order to add the pro-grade jar to the classpath as `-jar` does not support to `-classpath` flag.

Comment (garrettfreibott) [inline distribution/solr-distro/src/main/resources/bin/solr]: The same modifications mentioned above were made here for starting Solr as a background process. Out of the box, DDF starts solr this way.

Comment (ahoffer) [inline distribution/solr-distro/src/main/resources/bin/solr]: I recommend pulling the security manager args into their own variable and highlighting it. Something like:

```
# ******************************************************
# *****          DDF SECURITY MANAGER              *****
SEC_MAN_ARGS=-Djava.security.manager=net.sourceforge.prograde.sm.ProGradeJSM -Dpolicy.provider=net.sourceforge.prograde.policy.ProGradePolicy -Djava.security.policy==../security/solr-default.policy -classpath ../security/pro-grade-1.1.3.jar:start.jar org.eclipse.jetty.start.Main
# ******************************************************
```

**Comment by garrettfreibott** [comment]
build now

File: distribution/solr-distro/src/main/resources/bin/solr [hunk 2]
```diff
@@ -0,0 +1,2175 @@
...
+# checkout, and do not want to locally modify bin/solr.in.sh, can put
+# a customized include file at ~/.solr.in.sh.
+#
+# If you would rather configure startup entirely from the environment, you
+# can disable the include by exporting an empty SOLR_INCLUDE, or by
+# ensuring that no include files exist in the aforementioned search list.
+
+# ******************************************************
+# *****          DDF SECURITY MANAGER              *****
+SEC_MAN_ARGS="-Djava.security.manager=net.sourceforge.prograde.sm.ProGradeJSM -Dpolicy.provider=net.sourceforge.prograde.policy.ProGradePolicy -Djava.security.policy==../security/solr-default.policy -Dsolr.ssl.keystore=${SOLR_SSL_KEY_STORE} -Dsolr.ssl.truststore=${SOLR_SSL_TRUST_STORE} -classpath ../security/pro-grade-1.1.3.jar:start.jar org.eclipse.jetty.start.Main"
```
Comment (garrettfreibott) [inline distribution/solr-distro/src/main/resources/bin/solr]: This variable were added to the script file. See below comment for locations were the variables are being used.

File: distribution/solr-distro/src/main/resources/bin/solr [hunk 3]
```diff
@@ -0,0 +1,2175 @@
...
+            shift
+        ;;
+        --)
+            shift
+            break
+        ;;
+        *)
+            if [ "${1:0:2}" == "-D" ]; then
+              # pass thru any opts that begin with -D (java system props)
+
```
Comment (garrettfreibott) [inline distribution/solr-distro/src/main/resources/bin/solr]: `$SEC_MAN_ARGS` was added here replacing the `-jar start.jar` that was used previously.

Comment (garrettfreibott) [inline distribution/solr-distro/src/main/resources/bin/solr]: Same here for Solr running in the background - `$SEC_MAN_ARGS` was added here replacing the `-jar start.jar` that was used previously.

**Comment by garrettfreibott** [comment]
build now

**Pull request merged**